unified_planning/model/state.py

```diff
@@ -13,12 +13,108 @@
 # limitations under the License.
 #
 
+from typing import Dict, List, Optional
 import unified_planning as up
+from unified_planning.exceptions import UPUsageError, UPValueError
 
 
-class State:
-    '''This is an abstract class representing a classical state'''
+class ROState:
+    '''This is an abstract class representing a classical Read Only state'''
 
-    def get_value(self, f: 'up.model.FNode'):
-        '''Returns the value assignment for the given fluent expression'''
+    def get_value(self, value: 'up.model.FNode') -> 'up.model.FNode':
+        '''
+        This method retrieves the value in the state.
+        NOTE that the searched value must be set in the state.
+        :params value: The value searched for in the state.
+        :return: The set value.
+        '''
         raise NotImplementedError
+
+
+class RWState(ROState):
+    '''This is an abstract class representing a classical Read/Write state'''
+
+    def set_values(self, new_values: Dict['up.model.FNode', 'up.model.FNode']) -> 'RWState':
+        '''
+        Returns a different UPRWState in which every value in new_values.keys() is evaluated as his mapping
+        in new the new_values dict and every other value is evaluated as in self.
+        :param new_values: The dictionary that contains the values that need to be updated in the new State.
+        :return: The new State created.
+        '''
+        raise NotImplementedError
+
+
+class UPRWState(RWState):
```

Comment:
```suggestion
class UPRWState(RWState, ROState):
```
Right?

Reply:
Why? `RWState` inherits from `ROState`...

Reply:
You are right, I missed it